consumers/aws.go

```diff
@@ -54,17 +52,54 @@ func withClient(c AWSClient) *awsClient {
 }
 
 func (a *awsClient) Sync(endpoints []*pkg.Endpoint) error {
-	existingRecords, err := a.client.ListRecordSets()
+	newAliasRecords, err := a.client.EndpointsToAlias(endpoints)
 	if err != nil {
 		return err
 	}
 
-	recordInfoMap := a.client.RecordInfo(existingRecords)
-	newAliasRecords, err := a.client.EndpointsToAlias(endpoints)
+	hostedZonesMap, err := a.client.GetHostedZones()
+	if err != nil {
+		return err
+	}
+
+	inputByZoneID := map[string][]*route53.ResourceRecordSet{}
+	for _, record := range newAliasRecords {
+		zoneID := getZoneIDForEndpoint(hostedZonesMap, record) //this guarantees that the endpoint will not be created in multiple hosted zones
+		if zoneID == "" {
+			log.Infof("Hosted zone for endpoint: %s is not found. Skipping record...", record.Name)
```

Comment:
I don't know of any source of truth for that matter. I am guessing it makes sense to use to warning - when it leads to the outcome different from user expected (though it might make sense to use Info here as well). Error when the outcome is unexpected and crucial for program operation. Info to log general programming running stages/processes. Though we should probably discuss this more ;)